api/hasura/auth.ts

```diff
@@ -26,9 +26,26 @@ export default async function handler(req: VercelRequest, res: VercelResponse) {
       },
     });
     assert(profile, 'Profile cannot be found');
+    const users = await prisma.user.findMany({
+      where: {
+        address: profile.address,
+      },
+      select: {
+        id: true,
+        circle_id: true,
+        role: true,
+      },
+    });
     res.status(200).json({
       'X-Hasura-User-Id': tokenRow.tokenable_id.toString(),
       'X-Hasura-Role': profile.admin_view ? 'superadmin' : 'user',
+      'X-Hasura-Circle-User-Ids': `{ ${users
+        .map(u => u.id.toString())
+        .join()} }`,
+      'X-Hasura-Admin-Circle-Ids': `{ ${users
+        .filter(u => u.role === 1)
+        .map(u => u.circle_id.toString())
+        .join()} }`,
```

Comment:
This is one way to do it, but actually not necessary since all this info is available in the same database. [See here](https://hasura.io/docs/latest/graphql/core/auth/authorization/roles-variables.html#relationship-information-is-available-in-the-same-database)

This is fine for now as it might make it simpler / cleaner to define these permissions in the console instead of having to setup the complicated rule each time

Reply:
> it might make it simpler / cleaner to define these permissions in the console instead of having to setup the complicated rule each time

+1 -- more performant too, unless Hasura is really smart about recognizing & caching common subqueries